<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>888</width>
    <height>999</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>888</width>
    <height>999</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Hashui</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_10">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="topMargin">
       <number>8</number>
      </property>
      <property name="bottomMargin">
       <number>8</number>
      </property>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <property name="rightMargin">
         <number>8</number>
        </property>
        <item>
         <widget class="QLabel" name="labelTextEdit">
          <property name="font">
           <font>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Text input</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="textEdit"/>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="leftMargin">
         <number>8</number>
        </property>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QLabel" name="labelSHA1">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA1</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA1">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelSHA256">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA-256</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA256">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelSHA384">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA-384</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA384">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelSHA512">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA-512</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA512">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelFNV1a">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>FNV-1a</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditFNV1a">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelCRC8">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>CRC8</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditCRC8">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QLabel" name="labelSHA3_224">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA3-224</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA3_224">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelSHA3_256">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA3-256</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA3_256">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelSHA3_384">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA3-384</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA3_384">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelSHA3_512">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA3-512</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA3_512">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelMD5">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>MD5</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditMD5">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelCRC32">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>CRC32</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditCRC32">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="Line" name="line_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <property name="topMargin">
       <number>8</number>
      </property>
      <property name="bottomMargin">
       <number>8</number>
      </property>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <property name="rightMargin">
         <number>8</number>
        </property>
        <item>
         <widget class="QLabel" name="labelTextEdit_2">
          <property name="font">
           <font>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>lowercase</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="textEditLowercase">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line_3">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <property name="leftMargin">
         <number>8</number>
        </property>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QLabel" name="labelSHA1_lowercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA1</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA1_lowercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelSHA256_lowercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA-256</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA256_lowercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelSHA384_lowercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA-384</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA384_lowercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelSHA512_lowercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA-512</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA512_lowercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelFNV1a_lowercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>FNV-1a</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditFNV1a_lowercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelCRC8_lowercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>CRC8</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditCRC8_lowercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QLabel" name="labelSHA3_224_lowercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA3-224</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA3_224_lowercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelSHA3_256_lowercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA3-256</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA3_256_lowercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelSHA3_384_lowercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA3-384</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA3_384_lowercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelSHA3_512_lowercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA3-512</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA3_512_lowercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelMD5_lowercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>MD5</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditMD5_lowercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelCRC32_lowercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>CRC32</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditCRC32_lowercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="Line" name="line_5">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_5">
      <property name="topMargin">
       <number>8</number>
      </property>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <property name="rightMargin">
         <number>8</number>
        </property>
        <item>
         <widget class="QLabel" name="labelTextEdit_3">
          <property name="font">
           <font>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>UPPERCASE</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="textEditUppercase">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line_4">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <property name="leftMargin">
         <number>8</number>
        </property>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <widget class="QLabel" name="labelSHA1_uppercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA1</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA1_uppercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelSHA256_uppercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA-256</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA256_uppercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelSHA384_uppercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA-384</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA384_uppercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelSHA512_uppercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA-512</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA512_uppercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelFNV1a_uppercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>FNV-1a</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditFNV1a_uppercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelCRC8_uppercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>CRC8</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditCRC8_uppercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_9">
          <item>
           <widget class="QLabel" name="labelSHA3_224_uppercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA3-224</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA3_224_uppercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelSHA3_256_uppercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA3-256</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA3_256_uppercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelSHA3_384_uppercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA3-384</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA3_384_uppercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelSHA3_512_uppercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>SHA3-512</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditSHA3_512_uppercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelMD5_uppercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>MD5</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditMD5_uppercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelCRC32_uppercase">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>CRC32</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditCRC32_uppercase">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>888</width>
     <height>17</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
